lib/Interpreter/CIFactory.cpp

```diff
@@ -893,16 +902,64 @@ namespace {
     // of the two is the underlying compiler.
 
 #ifdef __clang__
-    PPOpts.addMacroDef("__CLING__clang__=" ClingStringify(__clang__));
+  #if defined(__apple_build_version__)
+     // __apple_build_version__ >= 8000031 Xcode 8
+     #if __apple_build_version__ >= 7030029
```

Comment:
Because of Apple's macro's there's no good way to get the version on clang compiled with.  It's a nice opportunity to be more specific, and this makes it much easier to get the version in cling. __clang__ is already defined as __clang__ so it's not losing any information.